spec/support/matchers/has_flash_notice.rb

```diff
@@ -0,0 +1,42 @@
+module Support
+  module Matchers
+    def have_flash_notice(text)
+      HasFlashNotice.new(text)
+    end
+
+    class HasFlashNotice
+      def initialize(text)
+        @text = text
+      end
+
+      def matches?(page)
+        @page = page
+
+        has_matching_row?
+      end
+
+      def description
+        "have flash notice with '#{text}'"
+      end
+
+      def failure_message
+        "Expected page to have flash notice with '#{text}', but it did not"
+      end
+
+      def failure_message_when_negated
+        "Expected page not to have flash notice with '#{text}', but it did"
+      end
+
+      private
+
+      attr_reader :page, :text
+
+      def has_matching_row?
+        page.within('.notice') do
```

Comment:
I am not sure how to do this. Should I create an additional matcher that will be invoked within `has_flash_notice`?